### PR TITLE
Fix git-fixup

### PR DESCRIPTION
**Pull request by MerelyAPseudonym**
TLDR: only the zsh-completion was getting installed, not the actual script.

```none
$ brew install git-fixup
==> Downloading https://homebrew.bintray.com/bottles/git-fixup-1.0.0.yosemite.bo
Already downloaded: /Library/Caches/Homebrew/git-fixup-1.0.0.yosemite.bottle.tar.gz
==> Pouring git-fixup-1.0.0.yosemite.bottle.tar.gz
==> Caveats
zsh completion has been installed to:
  /usr/local/share/zsh/site-functions
==> Summary
🍺  /usr/local/Cellar/git-fixup/1.0.0: 4 files, 16K
$ brew ls git-fixup
/usr/local/Cellar/git-fixup/1.0.0/share/zsh/site-functions/_git-fixup
```

There were two problems with the current formula.

1. The line
   ```ruby
   system "make", "prefix=#{prefix}", "install"
   ```
   should read
   ```ruby
   system "make", "PREFIX=#{prefix}", "install"
   ```
   The consequence of this mistake is that the actual `git-fixup` script is installed to `/usr/local/bin/git-fixup` instead of `/usr/local/Cellar/git-fixup/1.0.0/bin/git-fixup` since [`/usr/local` is the default value of `PREFIX`](https://github.com/keis/git-fixup/blob/628f13abc8f1dceff540053d833c642342524626/Makefile#L1).

2. **But** even if you change `prefix=#{prefix}` to `PREFIX=#{prefix}`, the installation will then *fail* because the necessary directories don't exist.
   ```none
   $ brew install git-fixup --interactive
   ==> Downloading https://github.com/keis/git-fixup/archive/v1.0.0.tar.gz
   Already downloaded: /Library/Caches/Homebrew/git-fixup-1.0.0.tar.gz
   ==> Entering interactive mode
   Type `exit' to return and finalize the installation
   Install to this prefix: /usr/local/Cellar/git-fixup/1.0.0
   bash-3.2$ make PREFIX=/usr/local/Cellar/git-fixup/1.0.0 install
   install -m755 git-fixup /usr/local/Cellar/git-fixup/1.0.0/bin/git-fixup
   install: /usr/local/Cellar/git-fixup/1.0.0/bin/git-fixup: No such file or directory
   make: *** [install] Error 71
   bash-3.2$ mkdir -p /usr/local/Cellar/git-fixup/1.0.0
   bash-3.2$ make PREFIX=/usr/local/Cellar/git-fixup/1.0.0 install
   install -m755 git-fixup /usr/local/Cellar/git-fixup/1.0.0/bin/git-fixup
   install: /usr/local/Cellar/git-fixup/1.0.0/bin/git-fixup: No such file or directory
   make: *** [install] Error 71
   bash-3.2$ mkdir -p /usr/local/Cellar/git-fixup/1.0.0/bin
   bash-3.2$ make PREFIX=/usr/local/Cellar/git-fixup/1.0.0 install
   install -m755 git-fixup /usr/local/Cellar/git-fixup/1.0.0/bin/git-fixup
   bash-3.2$ printf "$?\n"
   0 
```

### CODE REVIEW

**Comment by tdsmith** [comment]
Thanks for catching this and for the clear explanation! Can we do

```ruby
bin.mkpath
system "make", "PREFIX=...
```
instead, just in case `make install` gets more interesting later?

**Comment by mikemcquaid** [comment]
Agreed, we prefer `make install` whenever possible.

**Comment by MerelyAPseudonym** [comment]
@tdsmith You're welcome! And I made the changes, but it turns out git-fixup released 1.0.1, solving the problem.

@mikemcquaid Gotcha. I'll keep that in mind going forward.

**Comment by MerelyAPseudonym** [comment]
(And just let me know if I should squash the commits)

**Comment by tdsmith** [comment]
@BrewTestBot test this please

**Comment by tdsmith** [comment]
Thanks for the pull request! 🎉 Homebrew depends on contributions from community members like you and we're grateful for your support.
